rep-2011.rst

```diff
@@ -877,6 +877,22 @@ Similar to the integration with launch, you could run ``ros2 bag play ...`` and
 
 That option would handle the conversion on the fly using the same mechanisms as the command line tool.
 
+Security Concerns
+=================
+
+The additional features this REP proposes do not impose significantly increased security concerns, since the message conversions happen over topics and services.
+
+More precisely, transfer functions would just be another node that interfaces with topics and services to do the message conversions, and hence would be secured by the same security mechanisms that secure nodes, topics, and services in general.
+Furthermore, transfer functions would have to be defined ahead of time (local to the machine running the transfer function), and cycles in the transfer function chain would throw assertions, making it less feasible to use the transfer function feature as an attack vector.
```

Comment:
This confused me. Can you further elaborate on this consideration? Will transfer functions *always* be <ins>local to the machine running the transfer function</ins> and <ins>defined ahead of time</ins>? Wouldn't we be loosing *evolving* capabilities if this is case?